conf/homeserver.yaml

```diff
@@ -978,7 +978,7 @@ media_storage_providers:
 # your reverse proxy's config. Notably Nginx has a small max body size by default.
 # See https://matrix-org.github.io/synapse/latest/reverse_proxy.html.
 #
-max_upload_size: 100M
+max_upload_size: __MAX_UPLOAD_SIZE__
```

Comment:
You need to update also [this line](https://github.com/YunoHost-Apps/synapse_ynh/blob/testing/conf/nginx.conf#L7) too to have the correct value on both side.

Reply:
I asked if we can bind two keys in 2 different files in the config_panel.toml

Reply:
answer is NO. We need a custom setter to change the value in both files @thardev ;)

Reply:
should be something like:
```
set__max_upload_size() { 
    ynh_write_var_in_file --file=/etc/matrix-$app/homeserver.yaml --key=max_upload_size --value="${max_upload_size}"
    ynh_replace_string --match_string="max_upload_size .*" --replace_string="max_upload_size ${max_upload_size}" --target_file="/etc/nginx/conf.d/$domain.d/$app.conf"
    ynh_add_nginx_conf
}
```

Reply:
It look like good for me.

I'm not sure that ynh_replace_string work with regex. But if it don't works you can just use sed instead.

$I've tested this it should works:
```bash
sed -i -r "s|client_max_body_size\s+[[:digit:]]+M;|client_max_body_size ${max_upload_size}M;|g" "/etc/nginx/conf.d/$domain.d/$app.conf
```

Reply:
Yes, this looks good to me, I also tested the `sed` and works fine, so we could use that setter 👍

Reply:
is it relevant to let the user insert the unit ? 
in the help section add the text "<value> followed by [G|M|K] or none" 
and in the sed 

sed -i -r "s|client_max_body_size\s+[[:digit:]]+**[G|M|K]?**;|client_max_body_size **${max_upload_size}**;|g" "/etc/nginx/conf.d/$domain.d/$app.conf

Reply:
solved thanks to @rosbeef